provider/slack/index.ts

```diff
@@ -49,46 +54,236 @@ async function listAllChannels(client: WebClient): Promise<ChannelInfo[]> {
+    const relatedInformation: string[] = []
+
+    if (message.attachments) {
+        for (const attachmentInfo of message.attachments) {
+            const fileXML = xmlBuilder.build({
+                relatedInfo: attachmentInfo.text || '',
+            })
+            relatedInformation.push(fileXML)
+        }
+    }
+
+    if (message.files) {
+        for (const attachmentInfo of message.files) {
+            const fileXML = xmlBuilder.build({
+                relatedInfo: attachmentInfo.title || '',
+            })
+            relatedInformation.push(fileXML)
+        }
     }
+
+    const allRelatedInformation = relatedInformation.join('\n')
+    const messageInfo = xmlBuilder.build({
+        message: message.text || '',
+        allRelatedInformation: allRelatedInformation,
+    })
+    return messageInfo
 }
 
 async function fetchThreadMessages(
     client: WebClient,
     channelId: string,
     threadTs: string
 ): Promise<string | null> {
-    const allMessages = []
+    const allMessages: MessageElement[] = []
     try {
         let response = await client.conversations.replies({ channel: channelId, ts: threadTs })
-        let messages = response.messages as any[]
-        allMessages.push(...messages)
+        if (response.messages) {
+            allMessages.push(...response.messages)
+        }
         while (response.has_more) {
             const nextCursor = response.response_metadata?.next_cursor
             response = await client.conversations.replies({
                 channel: channelId,
                 ts: threadTs,
                 cursor: nextCursor,
             })
-            messages = response.messages as any[]
-            allMessages.push(...messages)
+            if (response.messages) {
+                allMessages.push(...response.messages)
+            }
         }
-        return allMessages?.map(msg => msg.text).join('\n\n')
+        return allMessages?.map(msg => getMessageInformation(msg)).join('\n\n')
     } catch (error) {
         return null
     }
 }
 
+function extractThreadTs(permalink: string | undefined): string | null {
+    if (!permalink) {
+        return null
+    }
+    const url = new URL(permalink)
+    const queryParams = new URLSearchParams(url.search)
+    const threadTs = queryParams.get('thread_ts')
+    return threadTs
+}
+
+function extractChannelId(slackUrl: string | undefined): string | null {
+    if (!slackUrl) {
+        return null
+    }
+    const pattern = /archives\/([A-Z0-9]+)\//
+    const match = slackUrl.match(pattern)
+    if (match) {
+        return match[1]
+    }
+    return null
+}
+
+// Function to search messages using slack api
+async function searchQuery(
+    client: WebClient,
+    query: string,
+    numResults = 5
+): Promise<Match[] | undefined> {
+    try {
+        const result = await client.search.messages({ query: query, count: numResults })
+        return result?.messages?.matches
+    } catch (error) {
+        return undefined
```

Comment:
Log a message to `console.error()` ?

Reply:
I think the console logs are not visible by default but can add this :)